replication-api/src/main/java/org/codice/ditto/replication/api/data/Metadata.java

```diff
@@ -0,0 +1,96 @@
+package org.codice.ditto.replication.api.data;
+
+import java.net.URI;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A wrapper for metadata objects received from {@link
+ * org.codice.ditto.replication.api.NodeAdapter}s.
+ */
+public interface Metadata {
+
+  /**
+   * A globally unique ID. This ID must be the ID of the metadata on the {@link
+   * org.codice.ditto.replication.api.NodeAdapter}.
+   *
+   * @return the id of this {@code Metadata}
+   */
+  String getId();
+
+  /**
+   * The type of the metadata being wrapped.
+   *
+   * @return the type
+   */
+  Class getType();
+
+  /**
+   * The original metadata being wrapped.
+   *
+   * @return the wrapped metadata
+   */
+  Object getRawMetadata();
+
+  /**
+   * @return the URI locating the metadata, or {@code null} if there is no resource associated with
+   *     this metadata
+   */
+  URI getResourceUri();
+
+  /** @param resourceUri the URI locating the metadata */
+  void setResourceUri(URI resourceUri);
+
+  /** @return the size of the {@link Resource} represented by the metadata */
+  long getResourceSize();
+
+  /** @param resourceSize the size of the {@link Resource} represented by this metadata */
+  void setResourceSize(long resourceSize);
+
+  /** @return the {@link Date} this metadata was last modified */
+  Date getMetadataModified();
+
+  /**
+   * @return the {@link Date} the {@link Resource} represented by this metadata was last modified,
+   *     or {@code null} if there is no resource associated with this metadata
+   */
+  Date getResourceModified();
+
+  /**
+   * @param resourceModified the {@link Date} the {@link Resource} represented by this metadata was
+   *     last modified
+   */
+  void setResourceModified(Date resourceModified);
+
+  /** @return a set of tags associated with the wrapped metadata */
+  Set<String> getTags();
```

Comment:
❓ Should we consider not returning collections in interfaces like that to avoid either exposing internal structures or be forced to clone or wrap the internal collection int an unmodifable one?